syft/pkg/cataloger/java/gradle_parser.go

```diff
@@ -0,0 +1,181 @@
+package java
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+	"unicode"
+
+	"github.com/anchore/syft/internal/log"
+
+	"github.com/anchore/syft/internal/file"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common"
+)
+
+// integrity check
+var _ common.ParserFn = parseJaveGradle
+
+func parseJaveGradle(_ string, reader io.Reader) ([]pkg.Package, []artifact.Relationship, error) {
+	var packages []pkg.Package
+
+	dependencyScanner := bufio.NewScanner(reader)
+
+	variables := make(map[string][]byte)
+
+	// get each 'dependency {}' section
+	onDependency := func(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
I see why this particular parser function is complex, mostly since the build.gradle file is groovy syntax and short of a groovy AST parser there isn't a straightforward way to extract the information in question. The split functions are pretty dense from a reading and understanding point of view, and I'm afraid to change them without a body of tests around the split functions themselves.

Assuming there isn't any shared state in the stack that is leveraged, can you extract out all of the split functions and put them individually under test?